### ISSUE TITLE
Bug in delegate.update for joins

### ISSUE THREAD

**Issue by iambenkay** [morintd/prismock#630]
Hello, I noticed that there is a bug in the update function when trying to update joined tables. For example:
This works without any issue:
```js
await prisma.user.update({
  where: {
    id: user.id,
  },
  data: {
    kycLevel: KycLevel.EMAIL,
  },
});
```
But when we try to update a joined table as in below:
```js
await prisma.user.update({
  where: {
    id: user.id,
  },
  data: {
    kycLevel: KycLevel.EMAIL,
    emailVerificationAttempt: {
      update: {
        verified: true,
      }
    }
  },
});
```
It fails with the below error:
```
Cannot read properties of undefined (reading 'id')
TypeError: Cannot read properties of undefined (reading 'id')
    at /Users/benjamin/Programs/gafe-service/node_modules/prismock/src/lib/operations/update.ts:34:27
    at Array.forEach (<anonymous>)
    at update (/Users/benjamin/Programs/gafe-service/node_modules/prismock/src/lib/operations/update.ts:33:24)
    at current.getItems.reduce.toUpdate (/Users/benjamin/Programs/gafe-service/node_modules/prismock/src/lib/operations/update.ts:216:66)
    at Array.reduce (<anonymous>)
    at updateMany (/Users/benjamin/Programs/gafe-service/node_modules/prismock/src/lib/operations/update.ts:208:39)
    at Object.updateMany (/Users/benjamin/Programs/gafe-service/node_modules/prismock/src/lib/delegate.ts:67:33)
    at /Users/benjamin/Programs/gafe-service/node_modules/prismock/src/lib/operations/update.ts:148:14
```
Note that error message is truncated.

I did some investigation and tracked down the issue to [this line](https://github.com/morintd/prismock/blob/8a7e38739498d5476266149d559cb872cb05f487/src/lib/operations/update.ts#L147)

Basically the prisma API does not expect the sub table to have a data property but the code assumes that it will have one so it causes the recursive code to crash.

Removing the `.data` will address this issue.

My temporary fix is to update both tables independently:
```js
await prisma.user.update({
  where: {
    id: user.id,
  },
  data: {
    kycLevel: KycLevel.EMAIL,
    emailVerificationAttempt: {
      update: {
        verified: true,
      }
    }
  },
});

await prisma.emailVerificationAttempt.update({
  where: {
    userId: user.id,
  },
  data: {
    verified: true,
  }
});
```

 but I would love for a consolidated solution.
 
 I can send in a PR fixing this issue if you want.

**Comment by iambenkay** [comment]
@morintd after further investigation it appears the issue is more subtle than I thought. The library only covers for the scenario of many to one relationships but fails in the case of one to one. I will make a PR to try and address the issue.

**Comment by morintd** [comment]
Hello @iambenkay , thanks a lot 🙏 I’ll have a look at your PR as soon as I can
